airflow/providers/amazon/aws/transfers/local_to_s3.py

```diff
@@ -0,0 +1,107 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from typing import Optional, Union
+
+from airflow.models import BaseOperator
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+
+
+class LocalFilesystemToS3Operator(BaseOperator):
+    """
+    Uploads a file from a local filesystem to Amazon S3.
+
+    :param filename: Path to the local file. Path can be either absolute
+            (e.g. /path/to/file.ext) or relative (e.g. ../../foo/*/*.csv). (templated)
+    :type filename: str
+    :param dest_key: The key of the object to copy to. (templated)
+
+        It can be either full s3:// style url or relative path from root level.
+
+        When it's specified as a full s3:// url, please omit dest_bucket.
+    :type dest_key: str
+    :param dest_bucket: Name of the S3 bucket to where the object is copied. (templated)
+
+        It should be omitted when `dest_key` is provided as a full s3:// url.
```

Comment:
Nit:  I'd really prefer to see some logic to enforce this rather than a comment.

Reply:
+1